src/SmartComponents/Applications/Applications.js

```diff
@@ -1,96 +1,94 @@
-import React, { Component } from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Skeleton, PageHeader, PageHeaderTitle, Main } from '@redhat-cloud-services/frontend-components';
-import { Stack, StackItem, Card, CardBody } from '@patternfly/react-core';
-import FormRender from '@data-driven-forms/react-form-renderer';
-import { formFieldsMapper, layoutMapper } from '@data-driven-forms/pf4-component-mapper';
-import registryDecorator from '@redhat-cloud-services/frontend-components-utilities/files/Registry';
 import { register } from '../../store';
 import reducers  from '../../store/reducers';
-import { getSchema, saveValues } from '../../actions';
+import { getSchema, saveValues, getConfig } from '../../actions';
+import { RenderForms } from '../../PresentationalComponents';
 
-@registryDecorator()
-class Applications extends Component {
-    constructor(props) {
-        super(props);
-        register(reducers);
-        props.getSchema(props.match.params.id);
-        this.state = {
-            appName: this.props.match.params.id
-        };
-    }
+export const getAppId = ({ params } = {}) => {
+    return params && params.id || '';
+};
 
-    renderForms(schemas, appName, loaded) {
-        return (
-            <Stack>
-                { loaded
-                    ?  schemas.map((schema, i) => (
-                        <StackItem key={ `settings-form-${i}` }>
-                            <Card>
-                                <CardBody>
-                                    <FormRender
-                                        formFieldsMapper={ formFieldsMapper }
-                                        layoutMapper={ layoutMapper }
-                                        schema={ schema }
-                                        onSubmit={ (values) => this.props.saveValues(appName, values) }
-                                        canReset
-                                    />
-                                </CardBody>
-                            </Card>
-                        </StackItem>
-                    ))
-                    : <StackItem>
-                        <Card>
-                            <CardBody>
-                                <Skeleton size='lg' />
-                            </CardBody>
-                        </Card>
-                    </StackItem>
-                }
-            </Stack>);
+const Applications = ({ appsConfig, saveValues, match, getSchema, getConfig, configLoaded, loaded, schema }) => {
+    const currApp = appsConfig && appsConfig[getAppId(match)] || getAppId(match);
+    let appName;
+    if (currApp.frontend) {
```

Comment:
This logic seems a bit odd to me. `currApp.title` could be present even if  `currApp.frontend` is falsy.
I think that something like this would be better:
`const appName = (currApp.frontend ? currApp.frontend.title : currApp.title ) || currApp`

Reply:
Good point, I'll change it to shortcutts so we get fallbacks by default
```JS
const appName = ((currApp.frontend && currApp.frontend.title) || currApp.title) || currApp;
```